src/black/linegen.py

```diff
@@ -1260,6 +1260,13 @@ def normalize_invisible_parens(
                 child, parens_after=parens_after, mode=mode, features=features
             )
 
+        # Fixes a bug where invisible parens are not properly wrapped around
+        # case blocks.
+        if isinstance(child, Node) and child.type == syms.case_block:
```

Comment:
You also need to gate these code changes on the preview style. Look for similar checks like `if Preview.fix_power_op_line_length in mode`. In the test you'll want to add `# flags: --preview`.

Reply:
Thanks for the guidance! I've made the modifications!